src/util/bip143.rs

```diff
@@ -263,13 +178,14 @@ impl<R: DerefMut<Target=Transaction>> SigHashCache<R> {
     /// }
     /// ```
     pub fn access_witness(&mut self, input_index: usize) -> &mut Vec<Vec<u8>> {
-        &mut self.tx.input[input_index].witness
+        self.cache.access_witness(input_index)
     }
 }
 
 #[cfg(test)]
 #[allow(deprecated)]
 mod tests {
+    use std::str::FromStr;
```

Comment:
I believe it should be
```suggestion
    use core::str::FromStr;
```

Reply:
AFAIK `std` in tests is OK.